### PR TITLE
doc: Add the Redpanda Cloud guide

### PR DESCRIPTION
**Pull request by bobbyiliev**
### Motivation

Adding the adjusted Redpanda Cloud guide for the Materialize binary.

### CODE REVIEW

File: doc/user/content/integrations/redpanda-cloud.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      FROM KAFKA BROKER '<redpanda-broker-url>' TOPIC '<topic-name>' WITH (
+          sasl_mechanisms = 'SCRAM-SHA-256',
+          security_protocol = 'SASL_SSL',
+          sasl_username = '<your-username>',
+          sasl_password = '<your-password>',
+          ssl_ca_location = '/secrets/ca.crt'
+      )
+      FORMAT BYTES;
+    CREATE MATERIALIZED VIEW <topic_name> AS
+      SELECT * FROM <redpanda_cloud>;
```
Comment (morsapaes) [inline doc/user/content/integrations/redpanda-cloud.md:92]:
```suggestion
```

Comment (morsapaes) [reply doc/user/content/integrations/redpanda-cloud.md:92]: This is missing the intermediate conversion view step, so might as well just cut it out!

**Review by morsapaes — CHANGES_REQUESTED** [review]
Thanks, @bobbyiliev! This is just missing the same edit to `doc/user/content/integrations/_index.md` as #14002. For backports, it's handy to cherry-pick the commit from `main` so all changes are picked up, and then remove whatever doesn't apply!

**Comment by bobbyiliev** [comment]
> Thanks, @bobbyiliev! This is just missing the same edit to `doc/user/content/integrations/_index.md` as #14002. For backports, it's handy to cherry-pick the commit from `main` so all changes are picked up, and then remove whatever doesn't apply!

Good catch! Just updated this.

**Review by morsapaes — APPROVED** [review]

**Pull request merged**